README.md

```diff
@@ -30,19 +30,45 @@ function defineMyModel(dynamo) {
       ranger: dynamo.types.timeUUID()
     }
   });
+  //
+  // A sync function must have a length of less than 2 if you are to be able
+  // to use the AwaitWrap wrapper
+  //
+  function exampleSyncFn() {
+    // do something sync
+    return someSyncResult;
+  }
+
+  function exampleAsyncFn(data, next) {
+    // do something async
+    next(null, someAsyncResult);
+  }
+
+  return new Dynastar({ model, hashKey: 'hashme', rangeKey: 'ranger', exampleSyncFn, exampleAsyncFn });
+}
+
+const mymodel = defineMyModel(require('dynamodb'));
+```
 
-  function exampleFunction1(){
-    // do something
-  } 
+### `AwaitWrap`
 
-  function exampleFunction2(){
-    // do something
-  } 
+If you would like to enable an `await`able model, we have a class for that.
 
-  return new Dynastar({ model, hashKey: 'hashme', rangeKey: 'ranger', exampleFunction1, exampleFunction2 });
-}
+Building on the previous example...
+
+```js
+const { AwaitWrap } = require('dynastar');
+
+const myAwaitModel = new AwaitWrap(mymodel);
+
+// In this circumstance we have a sync function and async function that was
+// added as "extra" onto the model itself. In this context the sync function
+// is left untouch but the async callback functon is made to be a `thenable`
+// that can be awaited
+
+const asyncResult = await myAwaitModel.exampleAsyncFn(data);
+const syncResult = myAwaitModel.exampleSyncFn(data);
```

Comment:
example nit:
you defined `exampleSyncFn` above as parameterless, either give that fn a parameter, or have this usage not have any parameters.